draft-ietf-quic-transport.md

```diff
@@ -1641,12 +1641,14 @@ of connection establishment.  By giving the client a different connection ID to
 use, a server can cause the connection to be routed to a server instance with
 more resources available for new connections.
 
-If a server receives a client Initial with an invalid Retry token,
-it knows the client will not accept another Retry token.  It can either
-proceed with the handshake without verifying the token or immediately close
-({{immediate-close}}) the connection with an connection error of
-INVALID_TOKEN to cause the handshake to fail quickly instead of waiting
-for the client to timeout.
+If a server receives a client Initial that can be unprotected but contains an
+invalid Retry token, it knows the client will not accept another Retry token.
+The server can drop such a packet and allow the client to time out to detect
+handshake failure, but that is a significant latency penalty to the client.
```

Comment:
Maybe instead:

> The server could discard such a packet, but this could result in the client continuing to send the invalid Initial packet until it is eventually forced to abandon the connection attempt.  A server that requires address validation SHOULD immediately close ({{imm...}}) the connection with an INVALID_TOKEN error to allow the client to detect the error sooner.  Note that a server has not established any state for the connection at this point and so does not enter the closing period.  Alternatively, a server MAY proceed with the connection without verifying the token, though the server MUST NOT consider the client address to be valid.

Reply:
I definitely like this text better. That said, I'm not sure that we should recommend sending a connection close. Esp. when under DoS, we do not want servers doing more work than necessary.

Reply:
What does Roberto say again?  I think that if the server is under duress, then dropping the packet is fine.  This is a recommendation that allows the server to help the client, that's all.  Do you think we need to acknowledge the possibility that the server is overloaded and add the third case?

Reply:
The text seems to assume that an implementer will handle the third case, where I was leaning towards being explicit. Perhaps something like: "When under duress, the server could discard such a packet. Dropping such a packet could however result in ..." That makes it a bit more explicit.

Reply:
I reworked Martin's suggestions, though I preferred to keep the sentence I had to his first sentence.  

In regards to "When under duress, the server could discard such a packet." I don't think that's adding anything on top of being clear about the consequences of discarding it, so I'd rather not add it.

Reply:
Discarding is mentioned up front, so we don't need to concern ourselves with repeating that.